tests/validation_tests/test_failures.py

```diff
@@ -47,3 +48,56 @@ def test_serialise_user_centered_failures():
             ]
         }
     }
+
+
+def test_non_nullable_user_centered_failures():
+    failure1 = NonNullableConstraintFailure(
+        sheet="Project Details",
+        column="Locations",
+    )
+    failure2 = NonNullableConstraintFailure(
+        sheet="Outcome_Data",
+        column="Unit of Measurement",
+    )
+    failure3 = NonNullableConstraintFailure(
+        sheet="Output_Data",
+        column="Unit of Measurement",
+    )
+
+    failures = [
+        failure1,
+        failure2,
+        failure3,
+    ]
+    output = serialise_user_centered_failures(failures)
+    assert output
+    assert "Project Admin" in output
+    assert "Project Details" in output["Project Admin"]
+    assert "Outcomes" in output
+    assert "Outcome Indicators (excluding footfall) / Footfall Indicator" in output["Outcomes"]
+    assert "Project Outputs" in output
+    assert "Project Outputs" in output["Project Outputs"]
+    assert len(output["Project Admin"]["Project Details"]) == 1
+    assert len(output["Outcomes"]["Outcome Indicators (excluding footfall) / Footfall Indicator"]) == 1
+    assert len(output["Project Outputs"]["Project Outputs"]) == 1
+    assert output["Project Admin"]["Project Details"]
+    assert output == {
+        "Project Admin": {
+            "Project Details": [
+                'There are blank cells in column: "Project Location(s) - Post Code (e.g. SW1P 4DF)". Use the space '
+                "provided to tell us the relevant information"
+            ]
+        },
+        "Outcomes": {
+            "Outcome Indicators (excluding footfall) / Footfall Indicator": [
+                'There are blank cells in column: "Unit of Measurement". Use the space provided to '
+                "tell us the relevant information"
+            ]
+        },
+        "Project Outputs": {
+            "Project Outputs": [
+                'There are blank cells in column: "Unit of Measurement". Use the space provided to '
+                "tell us the relevant information"
+            ]
+        },
+    }
```

Comment:
I think it will be most useful to just have a single assertion on the full output. The initial granularity is a lot of repetition that I don't think makes it easier to debug.